performant-software/dm-2
# Persistent project and text document creation/editing

### What does this PR do?
- Adds integration between front end redux store and back end rails API, for projects and documents
- Retrieves list of projects from API and provides links to them in home view list
- Updates URL and loads project upon selection
- Makes project and document titles editable
- Allows opening resources from table of contents and creating new text documents via button in table of contents
- Persists changes to content of text documents
- Persists highlights created in text documents
- Updates rails/node deployment approach to allow dynamic front end URLs and use yarn instead of npm in deployment

### What issues does it address?
- Closes #2 
- Closes #3 
- Closes #35 
- Closes #36

### How to test
- Go to the [PR review app](https://dm-2-staging-pr-34.herokuapp.com/)
- Create a new project using the 'Add Project' button at the bottom. You should now be in the project view and the URL should show the new project's id (note: no project deletion ability yet)
- Give the project a new title by clicking in the text field that says Untitled Project and changing it
- Return to the home view using the arrow icon next to the title. Your new project should appear in the list under 'Dummy Project'
- Return to your project by clicking its entry in the list
- Add a new text document by clicking Add New Document in the sidebar and choosing Text. A new text editor pane should appear in the main grid area.
- As with the project, edit the document's title in place. The title should update simultaneously in the summary for the document in the sidebar
- Add some content to the document using the text editor
- Add highlights on top of the text using the highlight-marker icon in the text editor menu. Use multiple colors by clicking the swatch element above that icon prior to adding a new highlight
- Clicking on an existing highlight should show an annotation popup with a summary of the highlight including its text content, highlighted in the appropriate color
- Try making multiple documents, closing them with the 'x' icon in the document viewer, reopening them by clicking in the sidebar, and deleting them using the trash icon. Any document currently open should have a blue backdrop to its summary in the sidebar; otherwise the backdrop should be gray
